File: app/src/main/java/com/se/masakapa/controllers/AuthController.java
package com.se.masakapa.controllers;

import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.se.masakapa.listeners.ListenerAdapter;
import com.se.masakapa.repositories.UserRepository;


public class AuthController {

    public static void Register(final String email, String password, final String displayName, final String profilePicture, final ListenerAdapter listener){
        // Validation here
        UserRepository.Register(email, password, new ListenerAdapter() {
            @Override
            public void newUserRegistered(FirebaseUser u) {
                listener.newItemInserted(UserRepository.InsertUser(u.getUid(), email, displayName, profilePicture));
            }
        });

    }
    public static Task<AuthResult> Login(String email, String password){
        return UserRepository.Login(email, password);
    }

    public static void Logout(){
        FirebaseAuth.getInstance().signOut();
    }
}

File: app/src/main/java/com/se/masakapa/controllers/AuthController.kt
package com.se.masakapa.controllers

import com.google.android.gms.tasks.Task
import com.google.firebase.auth.AuthResult
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import com.se.masakapa.listeners.Listeners
import com.se.masakapa.repositories.UserRepository

object AuthController {
    fun register(email: String?, password: String?, displayName: String?, profilePicture: String?, listener: Listeners) {
        // Validation here
        UserRepository.register(email, password, object : Listeners {
            override fun newUserRegistered(u: FirebaseUser) {
                listener.newItemInserted(UserRepository.insertUser(u.uid, email, displayName, profilePicture))
            }
        })
    }

    fun login(email: String?, password: String?): Task<AuthResult> {
        return UserRepository.login(email, password)
    }

    fun logout() {
        FirebaseAuth.getInstance().signOut()
    }
}